twisted/internet/interfaces.py

```diff
@@ -2207,6 +2207,28 @@ def clientConnectionForTLS(tlsProtocol):
 
 
 
+class IProtocolNegotiationFactory(Interface):
```

Comment:
Right, OK. So we already have a way to find out what it is, but we don't have a way to find out when it happens, but that's a completely separate issue - https://twistedmatrix.com/trac/ticket/6024 - which someone else should fix :)